src/formTimingTracking.js

```diff
@@ -10,30 +10,71 @@
     const getFieldName = (field) => field.getAttribute(fieldNameAttribute);
 
     const getFormName = (field) => field.closest("form").getAttribute(formNameAttribute);
+  
+    const trackFormSubmit = (e) => {
+        let formName = getFormName(e.target);
+        let fieldName = "Click";
+        let interactionType = "Submit";
+        subscribe({ formName, fieldName, interactionType });  
+    };
 
     const trackFormFieldEntry = (e) => {
         let fieldName = getFieldName(e.target);
         fieldsTimings[fieldName] = new Date().getTime();
     };
     
     const trackFormFieldLeave = (e) => {
-        let leaveType = e.type;
+        let interactionType = e.type;
         let formName = getFormName(e.target);
         let fieldName = getFieldName(e.target);
         // eslint-disable-next-line no-prototype-builtins
         if (fieldsTimings.hasOwnProperty(fieldName)) {
             let timeSpent = new Date().getTime() - fieldsTimings[fieldName];
             if (timeSpent > 0 && timeSpent < 1800000) {
-                subscribe({ formName, fieldName, leaveType, timeSpent });
+                subscribe({ formName, fieldName, interactionType, timeSpent });
             }
             delete fieldsTimings[fieldName];
         }
     };
 
-    document.querySelectorAll('input,select,textarea,datalist').forEach((elem) => {
+    const waitForElement = function(selector){
+            return new Promise(resolve => {
+                if (document.querySelector(selector)) {
+                    return resolve(document.querySelector(selector));
+                }
+
+                const observer = new MutationObserver(mutations => {
```

Comment:
`mutations` is not used, should be removed.

Reply:
`MutationObserver` is `heavy load` solution which might affect site performance, especially when `subtree` is true as I can see below. Maybe we should consider some solution based on internal form selector check? ew. setInterval every 100ms with its cancelation after success? @mgonera  what do you think?

Reply:
as RevOps we consider using MutationObserver a solution that is superior to using loops. Ok, it may be heavy load, but truly, the loops never cut it. There are always some edge cases, the loop cannot run indefinitely, and is not very friendly to handle. With MutationObserver we have one streamlined way of solving this type of cases. Such load times can be very different on various SPA types and we need a solution that accounts for all of them. Specifying a fixed time frame (e.g. with a timeout) never ends well, because those time frames will always differ.

Reply:
Ok if you accept that solution it is ok for me. PR Approved but @kbigos  please remember to remove unused parts of code before merge :)